free-3.md

```diff
@@ -1,43 +1,52 @@
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+A DSL for this seems pretty natural, as languages are often the best way to specify a new type of thing. Additionally, once this exists, a game object, or class, or data file can simply be queried in some familiar language syntax to get results. The domain can be very specific, as its literally just for specifying games and asking questions about them. 
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+Card games was on the list of suggested domains, and since I like card games a lot, I figured it would be interesting to think about DSLs that could make card games either easier or more interesting. This sort of would do both, as it would be a way to assist decision making, as well as consider interesting scenarios in an arbitrary game.
 
 ### Domain
 _Describe the project's domain in five words._
-
+Game specification and decision making
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+There would be two main interactions with the language. First, the user would write what a game looks like, probably using object oriented notation, as it makes the most sense for this domain. Second, the user would select a game, input all of the known information, and ask questions (like, my hand is (x,y,z), what should I play to maximize my chances of Q). 
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+When a program is run, it should basically do some simple AI techniques (probably just an expectimax search) with the parameters specified by the user in both the game and the current query. In terms of errors, the obvious case for an error is when there isn't enough information for the engine to run a search. In this case, there are two options. First, the program should tell the user this, and ask for any additional information. Second, if the user cannot give any, it should let the user know that it can't answer the question, and terminate "gracefully".
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+It should be as easy as possible to specify the game rules for whatever game you want. This has already been done, so I think its reasonable. It should also be as easy as possible to specify the current game state. Things that are possible but difficult probably include very advanced inference, where the user asks the game to make a decision based on his/her own strategies/playstyle. This would have to somehow be included in the values for different plays, so it would be hard to do, but theoretically possible.
 
```

Comment:
This part would be really tough, I think.  The amount of variety in the _types_ of rules that exist in card games is huge.  How would you notate, say, a card game that involves the ability to look at the opponent's hand (under some specific set of conditions)?